lib/spack/llnl/util/link_tree.py

```diff
@@ -62,8 +62,8 @@ def find_conflict(self, dest_root, ignore=None,
 
     def find_dir_conflicts(self, dest_root, ignore):
         conflicts = []
-        kwargs = {'follow_nonexisting': False, 'ignore': ignore}
-        for src, dest in traverse_tree(self._root, dest_root, **kwargs):
+        for src, dest in traverse_tree(self._root, dest_root, ignore=ignore,
+                                       follow_links=True, follow_nonexisting=False):
```

Comment:
Does `follow_links` need to be set `True` here after https://github.com/spack/spack/pull/29217/commits/bfc839ea07ac3dc797549a9a094841ecf1f5d195?

Reply:
Yeah, since it's called to test whether merge_directories / get_file_map+link will succeed, and the latter has `follow_links=True` so that symlinks to dirs are replaced with dirs. That way you can merge pkg_a with a symlink ./lib -> ./somewhere/lib, and pkg_b just has an actual directory ./lib.

Reply:
Yeah sorry to clarify: I mean can we set `follow_links=False` in *both* places as the default now that https://github.com/spack/spack/commit/bfc839ea07ac3dc797549a9a094841ecf1f5d195 is in this PR? In that case, we get the old behavior but, WRT the issue described in the PR description, `<prefix>/lib64` would now exist  (because we copy the symlink now).

Reply:
> because we copy the symlink now

I guess you mean yield? You would end up with just a directory `<prefix>/lib64` without any files in it if `follow_links=True` in `merge_directories` and `get_file_map`. That is exactly the buggy old behavior I'm trying to fix.

Reply:
Thanks, I was confused: I was thinking this could copy over the link as a link (like `copy_tree`), but that doesn't make any sense for `link_tree` since it needs to possibly merge several different `lib64` directories.

That being said, I'm concerned about moving to `follow_symlinks=True` by default without some additional logic to check for some potential problems:

* If the symlinks create a loop (e.g. a symlink refers to a parent directory)
* Symlinks that refer to one another in a manner that creates an exponential number of file paths for a given number of links

We haven't had to worry about either of those up to now since we don't follow the links in `link_tree`. Perhaps we could create some simple rules that would allow the case you discuss (or add a function which checks for these kinds of issues). This might seem paranoid but we are working with whatever the package installation has generated, so I think we should be careful.

Reply:
Actually, that's what this patch already does ;)